network/network.go

```diff
@@ -207,3 +218,49 @@ func (c *Collector) Collect(ch chan<- prometheus.Metric) {
 	scrapingElapsed := time.Since(scrapingStart)
 	c.logger.Log("level", "info", "message", "collected metrics", "scrapeID", c.scrapeID, "scrapeTime", scrapingElapsed.Seconds())
 }
+
+func (c *Collector) getNeighbours(n int, subsets []v1.EndpointSubset) ([]string, error) {
+	// Find our IP - note: this does not open a connection, due to UDP.
+	conn, err := net.Dial("udp", "8.8.8.8:80")
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	defer conn.Close()
+	ip := conn.LocalAddr().(*net.UDPAddr).IP.String()
```

Comment:
Don't we need a retry here because of the nature of UDP?

Reply:
Nope, because we're not actually sending any data.